File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/videos.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import OpenAiVideo, OpenAiVideoDeleteResponse, OpenAiVideoList, VideoGenerationRequest

class VideosApi:
    """videos API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def list_videos(self, params: Optional[Dict[str, Any]] = None) -> OpenAiVideoList:
        """List videos"""
        return self._client.get(f"/ai/v3/videos", params=params)

    def create_video(self, body: VideoGenerationRequest) -> OpenAiVideo:
        """Create video"""
        return self._client.post(f"/ai/v3/videos", json=body)

    def retrieve_video(self, videoId: str) -> OpenAiVideo:
        """Retrieve video"""
        return self._client.get(f"/ai/v3/videos/{videoId}")

    def delete_video(self, videoId: str) -> OpenAiVideoDeleteResponse:
        """Delete video"""
        return self._client.delete(f"/ai/v3/videos/{videoId}")

File: sdkwork-ai-sdk-flutter/lib/src/api/videos.dart
import '../http/client.dart';
import '../models.dart';

class VideosApi {
  final HttpClient _client;
  
  VideosApi(this._client);

  /// List videos
  Future<OpenAiVideoList?> listVideos(Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/videos'), params: params);
    return response is OpenAiVideoList ? response : null;
  }

  /// Create video
  Future<OpenAiVideo?> createVideo(VideoGenerationRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/videos'), body: body, contentType: 'application/json');
    return response is OpenAiVideo ? response : null;
  }

  /// Retrieve video
  Future<OpenAiVideo?> retrieveVideo(String video_id) async {
    final response = await _client.get(ApiPaths.aiPath('/videos/${video_id}'));
    return response is OpenAiVideo ? response : null;
  }

  /// Delete video
  Future<OpenAiVideoDeleteResponse?> deleteVideo(String video_id) async {
    final response = await _client.delete(ApiPaths.aiPath('/videos/${video_id}'));
    return response is OpenAiVideoDeleteResponse ? response : null;
  }
}
